src/convenience.jl

```diff
@@ -0,0 +1,186 @@
+stack[:relhum][Lat(Contains(-37), Lon(Contains(144))
+```
+"""
+function stack end
+
+stack(filename::AbstractString; kw...) = _constructor(stack, filename)(filename; kw...)
+function stack(filenames::NamedTuple; child_kwargs=(), kw...)
+    if all(x -> splitext(x)[2] == splitext(first(filenames))[2], filenames)
+        # All the same kind of file. We don't need to load them up front.
+        DiskStack(filenames; childtype=_constructor(geoarray, first(filenames)), kw...)
+    else
+        # These files are different extensions, just load them all
+        # as separate `AbstarctGeoArray` (which has some up front cost from
+        # reading the dimensions). They are probably still disk-backed for the
+        # actual array.
+        arrays = map(filenames) do fn
+            _constructor(geoarray, fn)(fn; child_kwargs...)
+        end
+        GeoStack(arrays; kw...)
+    end
+end
+
+"""
+    Base.write(filename::AbstractString, T::Type{<:AbstractGeoArray}, s::AbstractGeoStack)
+
+Write any [`AbstractGeoStack`](@ref) to file, guessing the backend from the file extension.
+
+Keyword arguments are passed to the `write` method for the backend.
+
+If the source can't be save as a stack-like object, individual array layers will be saved.
+"""
+function Base.write(filename::AbstractString, s::AbstractGeoStack; kw...)
+    base, ext = splitext(filename)
+    T = _constructor(stack, filename; throw=false)
+    if T isa Nothing
+        # Save as individual `geoarray`
+        T = _constructor(geoarray, filename)
+        for key in keys(s)
+            fn = joinpath(string(base, "_", key, ext))
+            write(fn, T, s[key])
+        end
+    else
+        write(filename, _constructor(stack, filename), s; kw...)
+    end
+end
+
+"""
+    series(dirpath::AbstractString, dims; kw...) => AbstractGeoSeries
+
+Load a vector of filepaths as a `AbstractGeoSeries`. `kw` are passed to the constructor.
+
+`dims` Dimensions can hold an index matching the files in the directory,
+or a function to convert the filename strings to index values.
+"""
+function series end
+
+function series(dirnames; 
+    child=geoarray, child_kwargs=nothing, window=(), kw...
+)
+    filepaths = readdir(path)
+    if all(x -> splitext(x)[2], filenames) == splitext(first(filenames))[2]
+        # All the same kind of file. We don't need to load them up front.
+        DiskStack(filenames; childtype=_get_constructor(geoarray, first(filenames)), kw...)
+    else
+        # These files are different extensions, just load them all
+        # as separate `AbstarctGeoArray` (which has some up front cost from
+        # reading the dimensions). They are probably still disk-backed for the actual array.
+        arrays = map(filenames) do dn
+            _constructor(geoarray, fn)(fn; window=window, child_kwargs...)
+        end
+        GeoStack(arrays; kw...)
+    end
+end
+
+# Support methods
+
+function _constructor(method::Function, filename; throw=true)
```

Comment:
I'd add a comment here to what this function does and why you do the in-direction.  But I probably use more comments than you do.

Reply:
Yep, I don't think I comment enough either.